src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java

```diff
@@ -248,52 +254,150 @@ public Optional<ShortCircuitAnalysisResultEntity> findResultsWithLimitViolations
+            }
+
+            return faultResultsPage;
+        } else {
+            return Page.empty();
+        }
+    }
+
+    private Pageable filterOutChildrenSort(Pageable pageable, Optional<Sort.Order> childrenSort) {
+        if (childrenSort.isEmpty()) {
+            return pageable;
         }
-        return faultResultsPage;
+        return PageRequest.of(
+                pageable.getPageNumber(),
+                pageable.getPageSize(),
+                Sort.by(pageable.getSort().stream().filter(sortOrder ->
+                                !sortOrder.getProperty().equals(childrenSort.get().getProperty())
+                        ).toList()
+                )
+        );
+    }
+
+    private Optional<Sort.Order> extractChildrenSort(Pageable pageable) {
+        return pageable.getSort().stream()
+                .filter(sortOrder ->
+                        sortOrder.getProperty().contains(FeederResultEntity.Fields.connectableId))
+                .findFirst();
     }
 
     @Transactional(readOnly = true)
     public Page<FeederResultEntity> findFeederResultsPage(ShortCircuitAnalysisResultEntity result, List<ResourceFilter> resourceFilters, Pageable pageable) {
         Objects.requireNonNull(result);
-        Specification<FeederResultEntity> specification = FeederResultSpecificationBuilder.buildSpecification(result.getResultUuid(), resourceFilters);
+        Specification<FeederResultEntity> specification = feederResultSpecificationBuilder.buildSpecification(result.getResultUuid(), resourceFilters);
         return feederResultRepository.findAll(specification, addDefaultSort(pageable, DEFAULT_FEEDER_RESULT_SORT_COLUMN));
     }
 
     @Transactional(readOnly = true)
-    public Page<FaultResultEntity> findFaultResultsWithLimitViolationsPage(ShortCircuitAnalysisResultEntity result, List<ResourceFilter> resourceFilters, Pageable pageable) {
+    public Page<FaultResultEntity> findFaultResultsWithLimitViolationsPage(ShortCircuitAnalysisResultEntity result,
+                                                                           List<ResourceFilter> resourceFilters,
+                                                                           Pageable pageable) {
         Objects.requireNonNull(result);
-        Specification<FaultResultEntity> specification = FaultResultSpecificationBuilder.buildSpecification(result.getResultUuid(), resourceFilters);
-        specification = FaultResultSpecificationBuilder.appendWithLimitViolationsToSpecification(specification);
+
+        Optional<Sort.Order> childrenSort = extractChildrenSort(pageable);
+
+        Pageable modifiedPageable = addDefaultSort(filterOutChildrenSort(pageable, childrenSort),
+                DEFAULT_FAULT_RESULT_SORT_COLUMN);
+        Specification<FaultResultEntity> specification = faultSpecBuilder.buildSpecification(result.getResultUuid(), resourceFilters);
+        specification = faultSpecBuilder.appendWithLimitViolationsToSpecification(specification);
```

Comment:
why not doing "appendWithLimitViolationsToSpecification" in "faultSpecBuilder.buildSpecification" directly ?